protoc_plugin/lib/indenting_writer.dart

```diff
@@ -101,8 +104,23 @@ class IndentingWriter {
     }
   }
 
+  void addSuffix(String suffixName, String text) {
+    suffixes[suffixName] = text;
+  }
+
   @override
-  String toString() => _buffer.toString();
+  String toString() {
+    if (suffixes.isNotEmpty) {
+      // TODO: We may want to introduce the notion of closing the file stream.
```

Comment:
This class doesn't have a file stream, what is the issue here? Should we open a GitHub issue for this?

Reply:
Sorry, I should have said `writer` here - the `IndentingWriter` class.

What I meant by this todo was that I'm mutating the state in the toString() method which isn't 100% clean behavior. It would be better to have a `close()` method on the writer, have people call that when finished generated code, and only emit the suffixes in close() and not in toString().

Reply:
We could also just rename `toString` to something else, maybe `finish` or something? In any case we can do it later.